packages/agents/src/AgentBase.ts

```diff
@@ -35,14 +35,20 @@ export abstract class AgentBase<TRunArgs, TAgentBaseContext extends AgentBaseCon
   }
 
   public async* run(
-    args: TRunArgs
+    args: TRunArgs,
+    context?: string
   ): AsyncGenerator<AgentOutput, RunResult, string | undefined> {
     const { chat } = this.context;
     try {
       this.config.initialMessages(args).forEach((message) => {
         chat.persistent(message.role, message.content);
       })
 
+      // If additional context is needed
+      if (context) {
+        chat.persistent("user", context);
```

Comment:
Kinda awkward, but fine for now

Reply:
Agreed, can improve in the future as we experiment with this feature.